strutil/strutil.go

```diff
@@ -111,3 +111,41 @@ func TruncateOutput(data []byte, maxLines, maxBytes int) []byte {
 	}
 	return data
 }
+
+// ParseValueWithUnit parses a value like 500kB and returns the number
+// in bytes. The case of the unit will be ignored for user convenience.
+func ParseValueWithUnit(inp string) (int64, error) {
+	unitMultiplier := map[string]int64{
+		"B": 1,
+		// strictly speaking this is "kB" but we ignore cases
+		"KB": 1000,
+		"MB": 1000 * 1000,
+		"GB": 1000 * 1000 * 1000,
+		"TB": 1000 * 1000 * 1000 * 1000,
+		"PB": 1000 * 1000 * 1000 * 1000 * 1000,
+		"EB": 1000 * 1000 * 1000 * 1000 * 1000 * 1000,
+	}
+	if len(inp) < 2 {
+		return 0, fmt.Errorf("cannot parse %q: need a unit", inp)
+	}
+	// most users will use a unit with two chars (e.g. "1024kB"),
+	// so check that first
+	unit := strings.ToUpper(inp[len(inp)-2:])
+	if mul, ok := unitMultiplier[unit]; ok {
+		val, err := strconv.ParseInt(inp[:len(inp)-2], 10, 64)
+		if err != nil {
+			return 0, err
+		}
+		return val * int64(mul), nil
+	}
+
+	// check for the simple case, e.g. "1234B"
+	if lastCh := inp[len(inp)-1]; lastCh != 'B' {
+		return 0, fmt.Errorf("cannot use unit in %q: try 'kB' or 'MB'", inp)
+	}
+	asByte, err := strconv.ParseInt(inp[:len(inp)-1], 10, 64)
+	if err != nil {
+		return 0, err
```

Comment:
This error will probably look awkward with a string like "200GiB".

Reply:
Nice catch, will fix.